### ISSUE TITLE
outlier_detection fails in calwebb_spec3 on ifu data

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#1814]
Running the calwebb_spec3 pipeline on an association of MIRI MRS (IFU) exposures results in the following error near the end of the outlier_detection step:
```
2018-03-15 18:08:10,833 - stpipe.Spec3Pipeline.outlier_detection - INFO - Time Create Single IFUcube  = 48.5.s
2018-03-15 18:08:10,930 - stpipe.CubeBuildStep - INFO - Number of IFUCube models returned from building single IFUCubes 3 
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    No attribute 'rfind'
Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/datamodels/properties.py", line 204, in __getattr__
    val = self._instance[attr]
KeyError: 'rfind'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 405, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec3.py", line 120, in process
    result = self.outlier_detection(result)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 405, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/outlier_detection/outlier_detection_step.py", line 158, in process
    step.do_detection()
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/outlier_detection/outlier_detection_ifu.py", line 111, in do_detection
    model, suffix=self.resample_suffix
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 829, in _make_output_path
    basename, basepath_ext = splitext(split(basepath)[1])
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/posixpath.py", line 103, in split
    i = p.rfind(sep) + 1
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.2a0.dev82-py3.5-macosx-10.7-x86_64.egg/jwst/datamodels/properties.py", line 207, in __getattr__
    raise AttributeError("No attribute '{0}'".format(attr))
AttributeError: No attribute 'rfind'
```
Appears to be some kind of error related to determining or constructing the output file name(s).

**Comment by hbushouse** [comment]
Example data (input _cal.fits and _asn.json files) can be found in /grp/jwst/ssb/bushouse/jwst_data/MIRI/MRS/kmurray/

**Comment by stsci-hack** [comment]
I am testing whether or not this error comes as a result of providing the wrong input to the make_output_path method; namely, a datamodel instead of a filename.

**Comment by stsci-hack** [comment]
@jemorrison should probably take a look at this test case as well, since I ran into another bug when this test data tried to run 'blot_cube' in outlier_detection after applying PR #1818.  The exact traceback was:

/user/hack/dev/release/stsci_python/pypi/jwst/git/jwst/jwst/outlier_detection/outlier_detection_ifu.py in do_detection(self)
    139             # each channel this will need to combine the multiple channels
    140             # of data into a single frame to match the original input...
--> 141             self.blot_median(median_model)
    142             if save_intermediate_results:
    143                 log.info("Writing out BLOT images...")

/user/hack/dev/release/stsci_python/pypi/jwst/git/jwst/jwst/outlier_detection/outlier_detection_ifu.py in blot_median(self, median_image)
    188         cubeblot = blot_cube_build.CubeBlot(median_image, self.input_models)
    189         cubeblot.blot_info()
--> 190         blot_models = cubeblot.blot_images()
    191 
    192         for j in range(len(blot_models)):

/user/hack/dev/release/stsci_python/pypi/jwst/git/jwst/jwst/cube_build/blot_cube_build.py in blot_images(self)
    131 
    132                 # get the detector values for this model
--> 133                 xstart, xend = instrument_info.GetMIRISliceEndPts(this_par1)
    134 #                xdet,ydet = wcstools.grid_from_bounding_box(model.meta.wcs.bounding_box, step=(1,1))
    135                 ydet, xdet = np.mgrid[:1024, :1032]

/user/hack/dev/release/stsci_python/pypi/jwst/git/jwst/jwst/cube_build/instrument_defaults.py in GetMIRISliceEndPts(self, parameter1)
    449 
    450     def GetMIRISliceEndPts(self, parameter1):
--> 451         slice_xstart = self.Info[parameter1]['xstart']
    452         slice_xend = self.Info[parameter1]['xend']
    453         return slice_xstart, slice_xend

KeyError: '12'

**Comment by jemorrison** [comment]
Ok I will take a look at it.

**Comment by jemorrison** [comment]
Closed this issue with #1842 . I tested it with MIRI data.
